karapace/backup/v3/schema.proto

```diff
@@ -0,0 +1,152 @@
+// The version 3 backup format of Karapace uses Protobuf serialization. This
+// data exchange format makes it possible to verify backward and forward
+// compatibility statically, and compared to a custom format enables us to add
+// new fields without introducing a new version.
+//
+// The backup file format itself is as follows:
+//
+// 1. The first byte is the “magic” marker, the NUL byte (`\x00`). This ensures
+//    that other tools reading the file know that this is a binary file (e.g.
+//    the `file` utility).[^1]
+// 2. The second byte is the backup format version, an unsigned 8-bit integer.
+//    This is always `\x03` for the backup format 3.[^1][^2]
+// 3. The next 4 bytes contain the byte count of the `Metadata` message that
+//    follows. The length is an unsigned 32-bit little-endian integer.[^3]
+// 4. The `Metadata` message.
+// 5. Zero or more records follow:
+//    1. 4 bytes containing the unsigned 32-bit little-endian integer byte count
+//       of the `Envelope` message that follows.[^3]
+//    2. The `Envelope` message.
+//
+// [^1]: The combination of these is referred to as the “magic marker version”.
+// [^2]: Endianness is irrelevant for a single byte integer.
+// [^3]: We are not using _varint_ encoding for the message framing, because it
+//   would require additional bound checks and logic to ensure that it actually
+//   is only a 32-bit value. Using fixed 4 bytes means that we can just read.
+//   Compression of the backups is used to reduce size, and we are not worried
+//   about the additional bytes required for encoding the framing bytes.
+syntax = "proto3";
+
+package karapace.backup.v3;
+
+// The metadata directly follows the magic marker version bytes. It contains
+// additional information regarding the backup.
+message Metadata {
```

Comment:
I'd like to have also the backup format version in the Metadata, not only in the header of the backup file.

Reply:
Asking for a duplicate. I edited the first comment as it was missing the reasoning, I am sorry as the original question was not clear.